### PR TITLE
Make cloud-init/userdata configurable

### PR DESCRIPTION
**Pull request by tboerger**
Currently the userdata is staticaly compiled in, with this change the
admin is able to customize the userdata or cloud-init content, the
content can be provided via path, base64 encoded string or a plain
string.

### CODE REVIEW

File: drivers/amazon/create.go [hunk 1]
```diff
@@ -10,9 +10,10 @@ import (
 	"encoding/base64"
 	"time"
 
-	"github.com/alecthomas/template"
 	"github.com/drone/autoscaler"
+	"github.com/drone/autoscaler/drivers/internal/scripts"
 
+	"github.com/alecthomas/template"
```
Comment (bradrydzewski) [inline drivers/amazon/create.go]: use text/template :)

File: drivers/amazon/provider.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 
 	"github.com/drone/autoscaler"
 
+	"github.com/alecthomas/template"
```
Comment (bradrydzewski) [inline drivers/amazon/provider.go]: text/template

**Comment by bradrydzewski** [comment]
looks good, can you replace `alecthomas/template` with `text/template`?  I had intended to use `text/template`, but goimports chose the wrong package.

File: drivers/amazon/create.go (hunk 1 above)
Comment (tboerger) [reply drivers/amazon/create.go]: Just followed your example :D

File: drivers/amazon/provider.go (hunk 2 above)
Comment (tboerger) [reply drivers/amazon/provider.go]: Just followed your example :D

**Comment by tboerger** [comment]
Replaced with text/template now.

**Pull request merged**